libraries/shared/storage/diff_repository.go

```diff
@@ -78,7 +78,7 @@ func (repository diffRepository) MarkChecked(id int64) error {
 func (repository diffRepository) GetFirstDiffIDForBlockHeight(blockHeight int64) (int64, error) {
 	var diffID int64
 	err := repository.db.Get(&diffID,
-		`SELECT id FROM public.storage_diff WHERE checked IS false AND block_height >= $1 LIMIT 1`, blockHeight)
+		`SELECT id FROM public.storage_diff WHERE block_height >= $1 LIMIT 1`, blockHeight)
 
```

Comment:
I figured that this method doesn't necessarily need to care if a diff is checked or not, since we're doing that anyway with `GetNewDiffs`. Also, was seeing some failures locally when `execute` caught up with all unchecked diffs, this method would fail because it didn't have an sql results.

This also got me thinking that we'll probably need to tweak this method and `MostRecentHeaderBlockNumber` for fresh deployed systems - again, will plan to look at that again tomorrow with fresh eyes.

Reply:
Nice catch! Makes me think maybe we want to handle `sql.ErrNoRows` upstream, since I imagine we could probably end up in this situation in other weird scenarios like if the interval is low and there aren't a lot of diffs being generated

Reply:
Was just checking out the [PR](https://github.com/makerdao/vdb-mcd-transformers/pull/184) updating the image and that looks legit - if we default to -1 but enable config, then fresh deployed systems can just not use this feature?

Reply:
yeah, that's a good idea. also, i added a guard in the watcher so that if either the headers or the storage_diff table returns a sql.NoRowsErr we just set the minID to 0, so at least the process won't fall over